### ISSUE TITLE
Could not find dynamic library

### ISSUE THREAD

**Issue by 1BE6-ankit** [mull-project/mull#528]
I have a docker setup for compiling my project. I am able to compile and run the project, but upon invoking mull I get the following error:
```
Extracting bitcode from executable (threads: 1): 1/1. Finished in 1ms.
Loading bitcode files (threads: 2): 2/2. Finished in 11ms.
Could not find dynamic library: libpthread.so.0
Could not find dynamic library: libstdc++.so.6
Could not find dynamic library: libm.so.6
Could not find dynamic library: libgcc_s.so.1
Could not find dynamic library: libc.so.6
Compiling instrumented code (threads: 2): 2/2. Finished in 10ms.
Loading dynamic libraries (threads: 1): 1/1. Finished in 0ms.
Searching tests (threads: 1): 1/1. Finished in 0ms.
Preparing original test run (threads: 1): 1/1. Finished in 3ms.
Running original tests (threads: 2): 0/2FactorialTest.NaturalCase failed: Crashed
FactorialTest.BaseCase failed: Crashed
Running original tests (threads: 2): 2/2. Finished in 191ms.
No mutants found. Mutation score: infinitely high

Total execution time: 218ms
```

My docker image looks like this:
```
FROM ubuntu:18.04

RUN mkdir -p /tmp/home
ENV HOME /tmp/home

RUN apt-get update && apt-get upgrade -y \
    && apt-get install -y git build-essential software-properties-common\
    && apt-get install -y vim wget

RUN apt-get update

RUN apt-get install -y python3-pip \
    && pip3 install conan

RUN apt-get update

## cpp unit
# RUN apt-get install -y autotools-dev automake libtool
RUN apt-get install -y libcppunit-dev  

# for mull
RUN apt-get install -y sqlite3 libsqlite3-dev libncurses5-dev libxml2-dev uuid-dev libstdc++-6-dev
```

To setup LLVM and mull I do the following:
```
export PATH="$PATH:/usr/src/llvm/bin"
cat > llvm.conf << "HEREDOC"
/usr/src/llvm/lib
HEREDOC

mv llvm.conf /etc/ld.so.conf.d/
ldconfig

export CC=/usr/src/llvm/bin/clang  
export CXX=/usr/src/llvm/bin/clang++

apt-get install -y cmake
apt-get update

dpkg -i setup/libicu55_55.1-7_amd64.deb
dpkg -i setup/Mull-0.3.0-LLVM-8.0-ubuntu-16.04.deb
```

**Comment by AlexDenisov** [comment]
You need to tell Mull where to look for the dynamic libraries. It does not make any assumptions in this regard.
You can use `ldd` to find the right places, e.g.:

```
> ldd your-binary
libpthread.so.0 => /usr/lib/libpthread.so.0 (0x00007feb9917a000)
libdl.so.2 => /usr/lib/libdl.so.2 (0x00007feb98f76000)
libstdc++.so.6 => /usr/lib/libstdc++.so.6 (0x00007feb98bf4000)
libm.so.6 => /usr/lib/libm.so.6 (0x00007feb988f6000)
libgcc_s.so.1 => /usr/lib/libgcc_s.so.1 (0x00007feb986e0000)
```

Then you need to add another option (one or more) to the `mull-cxx` invocation:

```
mull-cxx ... -ld_search_path /usr/lib
```

I hope this will fix the issue:

```
Running original tests (threads: 2): 0/2
FactorialTest.NaturalCase failed: Crashed
FactorialTest.BaseCase failed: Crashed
```

**Comment by 1BE6-ankit** [comment]
Thanks for the reply. By adding the -ld_search_path, dynamic libraries are loaded. But I am still getting the **Crashed** error. 

Is it because of my project setup: I am using conan to fetch gtest. My project structure is:
```
.
|-- CMakeLists.txt
|-- compile-build.sh
|-- conanfile.txt
|-- include
|   `-- factorial
|       `-- factorial.h
|-- src
|   |-- factorial.cpp
|   `-- main.cpp
`-- test
    `-- test-factorial.cpp
```

My CMakeLists.txt
```
cmake_minimum_required(VERSION 2.8.12)
project(MutationProj)

add_definitions("-std=c++11")

include(build/conanbuildinfo.cmake)
conan_basic_setup()

add_library(factorial-lib
    src/factorial.cpp)

# define headers
target_include_directories(factorial-lib
    PUBLIC 
        $<INSTALL_INTERFACE:include>    
        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
    PRIVATE
        ${CMAKE_CURRENT_SOURCE_DIR}/src
)

add_executable(factorial src/factorial.cpp src/main.cpp)
target_link_libraries(factorial factorial-lib)

# define tests
add_executable(test-factorial
    test/test-factorial.cpp)

target_link_libraries(test-factorial factorial-lib ${CONAN_LIBS})

add_test(test-factorial test-factorial)
```

I am able to compile and run the executables. Also, when I tested mull using the fmt example in the README, everything was working well.

**Comment by 1BE6-ankit** [comment]
The error turns out to be the result of incompatibility between the gtest version and mull. I was using 
_gtest/1.8.1@bincrafters/stable_, then I switched to _gtest/1.7.0@bincrafters/stable_. Upon switching, mull complained about not being able to load gtest dynamic libraries. But, after I linked them, every thing worked perfectly.

**Comment by AlexDenisov** [comment]
@1BE6-ankit thanks a lot for the debugging and nailing down the problem. Could please create a repository with the failing version? I would love to make it work for different versions of gtest and also include it into our integration test suite.

**Comment by 1BE6-ankit** [comment]
@AlexDenisov Here: https://github.com/1BE6-ankit/mull-conan-gtest

**Comment by AlexDenisov** [comment]
@1BE6-ankit I did a short debug session and here is what I've found so far:

The issue is not directly related to the GoogleTest, but rather to the way it is built by conan. I made your failing version work by enabling bitcode for google test (`export CXXFLAGS="-fembed-bitcode -g" && conan install ...`).

I think the original issue is related to this one #454.
I will continue investigating when I get more time.
